### DISCUSSION TITLE
is there a way to make unstable_getServerSession "reactive"

### DISCUSSION THREAD

**Discussion by mu6m** [nextauthjs/next-auth#6332] [Help]
### Question 💬

when using nextauth with next13 i tried making the session fetching process reactive 

i have a navbar component which needs the session so it can determine if the user is logged in or not 

But it seems that when using the next13 router for example

```
import { useRouter } from 'next/navigation';

export default function Page() {
  const router = useRouter();

  return (
    <button type="button" onClick={() => router.push('/dashboard')}>
      Dashboard
    </button>
  );
}
```

does not change the state of the nav bar

i tried using use react hook but still it does not work 

the only way that it works is when i try to refresh it manually through the browser 

i tried using the Session hook and it works but that's not what i wanted 

how can i make the unstable_getServerSession "reactive"

### How to reproduce ☕️

nav bar component

```
import SignOut from "@/components/sign-out";
import Link from "next/link";
import { use } from "react";
import { unstable_getServerSession } from "next-auth/next";

 async function userSes() {
  const ses = await unstable_getServerSession()
  return ses
}
export default function Nav() {
  const session = use(userSes());
  if(status === "loading"){
    return (
      <nav>
        <Link href="/">Home</Link>
      </nav>
    )
  }
  return (
    <nav>
        <Link href="/">Home</Link>
        {
          status === "authenticated"?
          <>
            <p>
              Signed in as {session?.user?.email}
            </p>
            <SignOut />
          </>:
          <>
              <Link href="/login">Login</Link>
              <Link href="/register">Register</Link>
          </>
        }
    </nav>
  );
}

```

sample useRouter component
```
import { useRouter } from 'next/navigation';

export default function Page() {
  const router = useRouter();

  return (
    <button type="button" onClick={() => router.push('/dashboard')}>
      Dashboard
    </button>
  );
}
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
It's not clear to me what the goal is, but server components only render initially, and such `unstable_getServerSession` is only invoked once. Then, the generated output is sent down to the client and hydrated.

**Reply by mu6m** [reply]
the problem that it only re-renders when i manually refresh the page
i want it to re-render when changing the route using 
```
router.push('/dashboard')
```
is there a way to do that
